### PR TITLE
fix: [CDS-80615]: Entity References for ServiceOverridesV2

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Describe your changes
Currently if a secret is used in Service OverridesV2 it does not show in secret references, with this change override entities will be shown in references.

Test Cases:
1) Created a global environment override of type secret, override was visible in that secrets references.
2) Updated a global environment override of type secret, override was visible in that secrets references and deleted from the previous secret.
3) Created a service specific override of type manifest, override was visible in connectors references.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)

### CODE REVIEW

**Comment by mohitg0795** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 208 at r4](https://reviewable.io/reviews/harness/harness-core/53942#-NgY5qaO0zFaS0U9NXCo:-NgY5qaO0zFaS0U9NXCp:buua5g3) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L208)):*
> ```Java
>       return serviceOverridesV2SetupUsageHelper.getAllReferredEntities(overridesEntity);
>     } catch (RuntimeException ex) {
>       throw new InvalidRequestException(
> ```

this shouldn't be invalidRequestException right? Should be something else.

**Review by mohitg0795 — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: 14 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 207 at r4](https://reviewable.io/reviews/harness/harness-core/53942#-NgY5wEAC7a6zUDM4Edc:-NgY5wEB-0iDZoQzbs_x:badiq2m) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L207)):*
> ```Java
>     try {
>       return serviceOverridesV2SetupUsageHelper.getAllReferredEntities(overridesEntity);
>     } catch (RuntimeException ex) {
> ```

shouldn't we log the exception here?

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: 14 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java` line 92 at r4](https://reviewable.io/reviews/harness/harness-core/53942#-NgY6Ci681ls2Qr-BE-K:-NgY6Ci681ls2Qr-BE-L:bh27ny1) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java#L92)):*
> ```Java
>   }
> 
>   public Set<EntityDetailProtoDTO> getAllReferredEntities(NGServiceOverridesEntity entity) {
> ```

lets try to group all public methods on the top for easy reference followed by all private and protected methods

File: 970-ng-commons/src/main/java/io/harness/ng/core/entitysetupusage/dto/EntityReferredByOverridesSetupUsageDetail.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.Builder;
+import lombok.Value;
+import org.springframework.data.annotation.TypeAlias;
+
+@OwnedBy(CDC)
+@Value
+@Builder
+@TypeAlias("EntityReferredByOverridesSetupUsageDetail")
+public class EntityReferredByOverridesSetupUsageDetail implements SetupUsageDetail {
+  String identifier;
```
Comment (tchaurasiya) [inline 970-ng-commons/src/main/java/io/harness/ng/core/entitysetupusage/dto/EntityReferredByOverridesSetupUsageDetail.java:23]: we should add scope of override, to make this unique

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java [hunk 2]
```diff
@@ -224,11 +251,12 @@ public Pair<NGServiceOverridesEntity, Boolean> upsert(@NonNull NGServiceOverride
     if (existingEntityInDb.isEmpty() && ServiceOverridesType.ENV_SERVICE_OVERRIDE == requestedEntity.getType()) {
       validateV1OverrideNotExistOrThrow(requestedEntity);
     }
-
+    Set<EntityDetailProtoDTO> referredEntities = getAndValidateReferredEntities(requestedEntity);
```
Comment (tchaurasiya) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java:254]: this should be after upsert operations, Else if this fails somehow user will not be able to create entity itself

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    final SetupUsageOwnerEntity ownerEntity = getOwnerEntity(entity);
+    if (isEmpty(referredEntities)) {
+      setupUsageHelper.deleteServiceSetupUsages(ownerEntity);
+    } else {
+      setupUsageHelper.publishOverridesEntitySetupUsage(ownerEntity, referredEntities, entity.getType().name());
+    }
+  }
+
+  /**
+   * Delete all setup usages where the 'referred by' is the current service entity
```
Comment (tchaurasiya) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java:59]: service entity??

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                                                                   .type(entity.getType())
+                                                                   .spec(entity.getSpec())
+                                                                   .environmentRef(entity.getEnvironmentRef())
+                                                                   .infraId(entity.getInfraIdentifier())
+                                                                   .build();
+    visitor.walkElementTree(serviceOverridesConfigV2);
+    return visitor.getEntityReferenceSet();
+  }
+
+  public Set<EntityDetailProtoDTO> getAllReferredEntities(NGServiceOverridesEntity entity) {
```
Comment (tchaurasiya) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java:92]: nit: can be make static final

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    } else {
+      setupUsageHelper.publishOverridesEntitySetupUsage(ownerEntity, referredEntities, entity.getType().name());
+    }
+  }
+
+  /**
+   * Delete all setup usages where the 'referred by' is the current service entity
+   */
+  public void deleteSetupUsages(NGServiceOverridesEntity entity) {
+    setupUsageHelper.deleteServiceSetupUsages(getOwnerEntity(entity));
```
Comment (tchaurasiya) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java:62]: why are we using serviceSetupUsageDelete Method, if this method is agnostic of entity type we should rename it.

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

**Review by sarthakkasat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: Some of the checks are too old. 14 file(s) to review. 2 change requests. 6. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 207 at r4](https://reviewable.io/reviews/harness/harness-core/53942#-NgY5wEAC7a6zUDM4Edc:-NgrEtdWBOUAqCwjs1-s:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L207)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

shouldn't we log the exception here?
</blockquote></details>

Done.
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 208 at r4](https://reviewable.io/reviews/harness/harness-core/53942#-NgY5qaO0zFaS0U9NXCo:-NgrEtyG6-N5hRMnDmTr:b-t2tb46) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L208)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

this shouldn't be invalidRequestException right? Should be something else.
</blockquote></details>

Could you please suggest what exception can we throw here, other methods to get referred entities are throwing the same error.
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 254 at r4](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360083575:-NgrF9o6CdqdrIluqccv:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L254)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

this should be after upsert operations, Else if this fails somehow user will not be able to create entity itself
</blockquote></details>

Done.
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java` line 59 at r4](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360086134:-NgrFAXpDIoEggchBQ5N:bbb5b9g) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java#L59)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

service entity??
</blockquote></details>

Update the text to override entity
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java` line 62 at r4](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360087385:-NgrFFtsDADuW7tSS_tQ:bw0jje8) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java#L62)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

why are we using serviceSetupUsageDelete Method, if this method is agnostic of entity type we should rename it.
</blockquote></details>

Update the name to deleteEntitySetupUsages
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java` line 92 at r4](https://reviewable.io/reviews/harness/harness-core/53942#-NgY6Ci681ls2Qr-BE-K:-NgrFLX8D6lPsBasspAD:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/setupusage/ServiceOverridesV2SetupUsageHelper.java#L92)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

lets try to group all public methods on the top for easy reference followed by all private and protected methods
</blockquote></details>

Done
___
*[`970-ng-commons/src/main/java/io/harness/ng/core/entitysetupusage/dto/EntityReferredByOverridesSetupUsageDetail.java` line 23 at r4](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360081354:-NgrFP0nDi327zA4iI1m:b7qtq0g) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/970-ng-commons/src/main/java/io/harness/ng/core/entitysetupusage/dto/EntityReferredByOverridesSetupUsageDetail.java#L23)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

we should add scope of override, to make this unique
</blockquote></details>

Added orgId, projectID and accountID to make it unique.

**Comment by mohitg0795** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 208 at r4](https://reviewable.io/reviews/harness/harness-core/53942#-NgY5qaO0zFaS0U9NXCo:-NgrNVnj8mPd_fBUyHG1:b-n5k4hg) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L208)):*
<details><summary><i>Previously, sarthakkasat (Sarthak Kasat) wrote…</i></summary><blockquote>

Could you please suggest what exception can we throw here, other methods to get referred entities are throwing the same error.
</blockquote></details>

InternalServerErrorException

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/NGServiceOverrideConfigV2.java [hunk 6]
```diff
@@ -30,11 +35,19 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 @FieldDefaults(level = AccessLevel.PRIVATE)
 @TypeAlias("serviceOverrideConfig")
-public class NGServiceOverrideConfigV2 {
-  @NonNull String identifier;
+@SimpleVisitorHelper(helperClass = NGServiceConfigVisitorHelper.class)
```
Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/NGServiceOverrideConfigV2.java:38]: Visitor should return this class object, We need to use another visitor.

Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/NGServiceOverrideConfigV2.java:38]: I am also not sure, How dummy visting node is used, Lets confirm with @Archit/ @Yogesh if this vistor is even needed.

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverridesSpec.java [hunk 7]
```diff
@@ -37,11 +42,30 @@
 @FieldDefaults(level = AccessLevel.PRIVATE)
 @JsonIgnoreProperties(ignoreUnknown = true)
 @Schema(name = "ServiceOverrideSpec", description = "This is the Service Override Spec entity defined in Harness")
+@SimpleVisitorHelper(helperClass = ServiceDefinitionVisitorHelper.class)
```
Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverridesSpec.java:45]: Visitor should return this class object, We need to use another visitor.

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverridesSpec.java [hunk 8]
```diff
@@ -37,11 +42,30 @@
...
-public class ServiceOverridesSpec {
+public class ServiceOverridesSpec implements Visitable {
   List<NGVariable> variables;
   List<ManifestConfigWrapper> manifests;
   List<ConfigFileWrapper> configFiles;
   ApplicationSettingsConfiguration applicationSettings;
   ConnectionStringsConfiguration connectionStrings;
+
+  @Override
+  public VisitableChildren getChildrenToWalk() {
```
Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverridesSpec.java:55]: we should add applicationSettings & applicationSettings too , they to have connectors

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

**Review by sarthakkasat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: 17 file(s) to review. 2 change requests. 9. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/NGServiceOverrideConfigV2.java` line 38 at r6](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360672296:-NguP3dv3T1ypPJ6B2Br:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3f30c50d7b451e4832d02bd84951681faef08640/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/NGServiceOverrideConfigV2.java#L38)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Visitor should return this class object, We need to use another visitor.
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/NGServiceOverrideConfigV2.java` line 38 at r6](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360673591:-NguP46E3nMMCc1PR7Zj:b-47ix6i) ([raw file](https://github.com/harness/harness-core/blob/3f30c50d7b451e4832d02bd84951681faef08640/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/NGServiceOverrideConfigV2.java#L38)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

I am also not sure, How dummy visting node is used, Lets confirm with @Archit/ @Yogesh if this vistor is even needed.
</blockquote></details>

Sure will discuss this
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverridesSpec.java` line 45 at r6](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360673882:-NguP7hn-JFO_ybUquYX:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3f30c50d7b451e4832d02bd84951681faef08640/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverridesSpec.java#L45)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Visitor should return this class object, We need to use another visitor.
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverridesSpec.java` line 55 at r6](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360676189:-NguP86I-yn_feu9jLcc:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3f30c50d7b451e4832d02bd84951681faef08640/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverridesSpec.java#L55)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

we should add applicationSettings & applicationSettings too , they to have connectors
</blockquote></details>

Done.

**Review by tchaurasiya — APPROVED** [review]

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]
> Reviewed 1 of 8 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: 17 file(s) to review. 2 change requests. 9. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`953-events-api/src/main/proto/io/harness/eventsframework/schemas/entitysetupusage/entity_detail_with_setup_usage_detail.proto` line 54 at r8](https://reviewable.io/reviews/harness/harness-core/53942#-NhAYYKW0irSSZPHOVMH:-NhAYYKW0irSSZPHOVMI:b-8sl6vj) ([raw file](https://github.com/harness/harness-core/blob/23e9020889d9a4e81d13a4075e78e63b39305b3f/953-events-api/src/main/proto/io/harness/eventsframework/schemas/entitysetupusage/entity_detail_with_setup_usage_detail.proto#L54)):*
> ```proto
>     string accountId = 9;
>     string orgId = 10;
>     string projectId = 11;
> ```

these numbers should be 1-5
___
*[`970-ng-commons/src/main/java/io/harness/ng/core/entitysetupusage/dto/EntityReferredByOverridesSetupUsageDetail.java` line 23 at r4](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360081354:-NhAY-qICqSse5lj8kvm:bokc7uk) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/970-ng-commons/src/main/java/io/harness/ng/core/entitysetupusage/dto/EntityReferredByOverridesSetupUsageDetail.java#L23)):*
<details><summary><i>Previously, sarthakkasat (Sarthak Kasat) wrote…</i></summary><blockquote>

Added orgId, projectID and accountID to make it unique.
</blockquote></details>

maybe put @NotNull on some fields

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 213 at r8](https://reviewable.io/reviews/harness/harness-core/53942#-NhA_APMATBF20gtI7CY:-NhA_APMATBF20gtI7CZ:bxvai5l) ([raw file](https://github.com/harness/harness-core/blob/23e9020889d9a4e81d13a4075e78e63b39305b3f/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L213)):*
> ```Java
>           ex);
>       throw new InternalServerErrorException(
>           String.format("Exception while retrieving referred entities for infrastructure: [%s]. ",
> ```

typo , infrastructure -> overrides ?

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 2 of 8 files at r5, 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: 17 file(s) to review. 2 change requests. 9. unresolved discussions. 1 of 2 approvals obtained.

**Review by sarthakkasat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: 17 file(s) to review. 2 change requests. 11. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java` line 213 at r8](https://reviewable.io/reviews/harness/harness-core/53942#-NhA_APMATBF20gtI7CY:-NhAoJFi5O9QRGdS-8mv:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/23e9020889d9a4e81d13a4075e78e63b39305b3f/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverridesServiceV2Impl.java#L213)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

typo , infrastructure -> overrides ?
</blockquote></details>

Done.
___
*[`953-events-api/src/main/proto/io/harness/eventsframework/schemas/entitysetupusage/entity_detail_with_setup_usage_detail.proto` line 54 at r8](https://reviewable.io/reviews/harness/harness-core/53942#-NhAYYKW0irSSZPHOVMH:-NhAoKB7EN4XjYbriTo8:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/23e9020889d9a4e81d13a4075e78e63b39305b3f/953-events-api/src/main/proto/io/harness/eventsframework/schemas/entitysetupusage/entity_detail_with_setup_usage_detail.proto#L54)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

these numbers should be 1-5
</blockquote></details>

Done.
___
*[`970-ng-commons/src/main/java/io/harness/ng/core/entitysetupusage/dto/EntityReferredByOverridesSetupUsageDetail.java` line 23 at r4](https://reviewable.io/reviews/harness/harness-core/53942#gh-1360081354:-NhAoL4E7AX4LKkchTqX:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/707c8db2024171b3533b739b59b5db44bc428008/970-ng-commons/src/main/java/io/harness/ng/core/entitysetupusage/dto/EntityReferredByOverridesSetupUsageDetail.java#L23)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

maybe put @NotNull on some fields
</blockquote></details>

Done.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 13 files at r1, 2 of 8 files at r5, 7 of 8 files at r7, 1 of 1 files at r8, 3 of 3 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: 17 file(s) to review. 1 change requests. 9. unresolved discussions. 1 of 2 approvals obtained.

**Comment by sarthakkasat** [comment]
trigger uts

**Review by tchaurasiya — DISMISSED** [review]

**Comment by sarthakkasat** [comment]
trigger review

**Comment by sarthakkasat** [comment]
trigger uts

**Comment by sarthakkasat** [comment]
trigger ut3, ut5

**Comment by sarthakkasat** [comment]
trigger ut3

**Comment by sarthakkasat** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53942)* status: 18 file(s) to review. 1 change requests. 9. unresolved discussions. 1 of 2 approvals obtained.

**Comment by sarthakkasat** [comment]
trigger ut5, ut6, ut7

**Comment by sarthakkasat** [comment]
trigger ut6

**Review by sourabh-awashti — APPROVED** [review]

**Comment by sarthakkasat** [comment]
trigger review

**Pull request merged**